index.js

```diff
@@ -44,9 +44,13 @@ function listToArray(list) {
   if (isString(list) && list !== '') {
     return stringToArray(list);
   } else if (list && list.length) {
-    return list.reduce(function (array, string) {
-      return array.concat(stringToArray(string));
-    }, []);
+    return list
+      .filter(function (string) {
+        return !!string;
+      })
+      .reduce(function (array, string) {
+        return array.concat(stringToArray(string));
+      }, []);
```

Comment:
While that would be more concise, I think having it in two steps is easier to understand.